eth/db/chain_gaps.py

```diff
@@ -12,16 +16,74 @@ class GapChange(enum.Enum):
     NewGap = enum.auto()
     GapFill = enum.auto()
     GapSplit = enum.auto()
-    GapShrink = enum.auto()
+    GapLeftShrink = enum.auto()
+    GapRightShrink = enum.auto()
     TailWrite = enum.auto()
 
 
-GAP_WRITES = (GapChange.GapFill, GapChange.GapSplit, GapChange.GapShrink)
+GAP_WRITES = (
+    GapChange.GapFill,
+    GapChange.GapSplit,
+    GapChange.GapLeftShrink,
+    GapChange.GapRightShrink,
+)
 GENESIS_CHAIN_GAPS = ((), BlockNumber(1))
 
 GapInfo = Tuple[GapChange, ChainGaps]
 
 
+@to_tuple
+def _join_overlapping_gaps(unjoined_gaps: Tuple[BlockRange, ...]) -> Iterable[BlockRange]:
+    """Input must already be sorted"""
```

Comment:
*Nit*: The docstring could probably be a bit more descriptive regarding what exactly this function does.